### PR TITLE
[ok_http]: Use the Android SDK to generate JNI bindings.

### PR DESCRIPTION
**Pull request by Anikate-De**
Items in this PR:
 - [x] Added `okhttp:4.12.0` dependency in `/example/android/app/build.gradle` (used for JNI), and updated `jnigen.yaml` to use the Android SDK instead.
 - [x] Remove `jar/` exclusion from `.gitignore`

**WHY?** As multiple Kotlin files with intermediary code will be written to enhance the usage of `OkHttp`, it is inconsiderable to push JAR files for future commits or to run an additional step of creating said JARs before generating the bindings.

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/ok_http/android/build.gradle [hunk 1]
```diff
@@ -23,12 +26,21 @@ rootProject.allprojects {
...
     if (project.android.hasProperty("namespace")) {
         namespace = "com.example.ok_http"
     }
 
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
+
+    sourceSets {
+        main.java.srcDirs += 'src/main/kotlin'
```
Comment (brianquinlan) [inline pkgs/ok_http/android/build.gradle:41]: Do you need this when you don't have any sources?

File: pkgs/ok_http/android/build.gradle [hunk 2]
```diff
@@ -23,12 +26,21 @@ rootProject.allprojects {
...
 apply plugin: "com.android.library"
+apply plugin: 'kotlin-android'
 
 android {
     if (project.android.hasProperty("namespace")) {
         namespace = "com.example.ok_http"
     }
 
+    kotlinOptions {
+        jvmTarget = '1.8'
```
Comment (brianquinlan) [inline pkgs/ok_http/android/build.gradle:37]: Do you need this when you don't have any sources?

File: pkgs/ok_http/android/build.gradle (hunk 1 above)
Comment (Anikate-De) [reply pkgs/ok_http/android/build.gradle:41]: I added them just as a pre-check to include all sources (next PR related to `testRedirects` will have Kotlin sources)

Should I remove them for now, and include in the next one?

Comment (brianquinlan) [reply pkgs/ok_http/android/build.gradle:41]: Yes, please add this to the PR that adds Kotlin code.

File: pkgs/ok_http/android/build.gradle (hunk 2 above)
Comment (brianquinlan) [reply pkgs/ok_http/android/build.gradle:37]: Yes, please add this to the PR that adds Kotlin code.

File: pkgs/ok_http/android/build.gradle (hunk 1 above)
Comment (Anikate-De) [reply pkgs/ok_http/android/build.gradle]: Removed ✅

File: pkgs/ok_http/android/build.gradle (hunk 2 above)
Comment (Anikate-De) [reply pkgs/ok_http/android/build.gradle]: Removed ✅

**Review by brianquinlan — APPROVED** [review]

**Pull request merged**